### DISCUSSION TITLE
Gaps between perimeter's corners

### DISCUSSION THREAD

**Discussion by filippods** [supermerill/SuperSlicer#2818] [General]
Hello, using the correct PA value the prints have gaps between perimeter's corners. Lowering the PA value there is overextrusion in direction changes, this results in more evident overextrusion in small solid infill areas.
Enabling "enforce 100% infill" option, the overxtrusion in small solid infill areas became more evident.

How can i solve this issue?

CORRECT PA
![20220505_172241~2.jpg](https://user-images.githubusercontent.com/8654901/168708073-4c86e349-d3e2-4cdd-b2a2-45ffef7fba3c.jpg)

LOWER PA
![20220505_201216~2.jpg](https://user-images.githubusercontent.com/8654901/168708296-3a87ae9c-5392-4611-8c1b-ce35f1363daa.jpg)

Thank you!!

**Comment by supermerill** [comment]
are you sure you don't under-extrude?
Is the weight of your objects the same as the one predicted by the slicer?

**Reply by filippods** [reply]
Hi, first of all, thank you for your hard work!
I haven't weighted the printed parts cause i don't know the density of my filament (Esun ABS+), however i've just tried to increase the EM, but all the part gets overextruded so i can exclude a miscalibrated EM.

**Reply by supermerill** [reply]
if it's abs, it may be unavoidable, as the abs can't flow easily...
but in your first picture, I can see the width of the lines vary (it became thinner near the voids), why is that?

**Reply by filippods** [reply]
Could you highlight the issue are you referring in the picture?

**Reply by supermerill** [reply]
![image](https://user-images.githubusercontent.com/6536403/169056273-afda0d60-3436-43af-9c7c-8646188297d0.png)

**Reply by filippods** [reply]
Ok, this is exactly what i'm referring. The PA value is well calibrated, but the filmant has this strange behaviour.

**Comment by cyr123** [comment]
You could try using a smaller PA smoothing time (I assume you are using klipper).
It looks to me just like the kind of under-extrusion you can get near corners because of PA "putting on the brakes" early.

https://klipper.discourse.group/t/pressure-advance-smooth-time-on-direct-extruders-with-short-filament-path/1971

**Reply by filippods** [reply]
Thank you cyr, my PA smooth time is 0.015.
